ui/app/models/keymgmt/provider.js

```diff
@@ -104,18 +109,48 @@ export default class KeymgmtProviderModel extends Model {
   }
 
   async fetchKeys(page) {
-    try {
-      this.keys = await this.store.lazyPaginatedQuery('keymgmt/key', {
-        backend: 'keymgmt',
-        provider: this.name,
-        responsePath: 'data.keys',
-        page,
-      });
-    } catch (error) {
-      this.keys = [];
-      if (error.httpStatus !== 404) {
-        throw error;
+    if (this.canListKeys) {
+      try {
+        this.keys = await this.store.lazyPaginatedQuery('keymgmt/key', {
+          backend: 'keymgmt',
+          provider: this.name,
+          responsePath: 'data.keys',
+          page,
+        });
+      } catch (error) {
+        this.keys = [];
+        if (error.httpStatus !== 404) {
```

Comment:
Gut check: This is because the response is a 404 if the provider has no distributed keys, correct?

Reply:
As far as I can remember yes :slightly_smiling_face:. That was added in phase 1 but from what I recall if the provider doesn't have any associated keys then we just want to update `this.keys` rather than surfacing an error.